src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java

```diff
@@ -48,14 +62,36 @@ public class IoTCoreClient implements MessageClient {
         }
     };
 
+    private final MqttClientConnectionEvents connectionCallbacks = new MqttClientConnectionEvents() {
+        @Override
+        public void onConnectionInterrupted(int errorCode) {
+        }
+
+        @Override
+        public void onConnectionResumed(boolean sessionPresent) {
+            synchronized (subscribeFuture) {
```

Comment:
what if this is null?

Reply:
Synchronizing on different objects will not make this safe. When manipulating `subscribeFuture` you must synchronize on the same object in every place that you are messing with it otherwise you're not accomplishing anything.

Reply:
updated to synchronize on enclosing object:
```
synchronized (this) {}
```